frontend/store/gas.ts

```diff
@@ -0,0 +1,102 @@
+import { ActionContext } from 'vuex';
+import type { GasParameters, TransactionFees } from 'auctions-core/src/types';
+import { getBaseFeePerGas, getGasParametersForTransaction } from 'auctions-core/src/gas';
+import BigNumber from 'bignumber.js';
+import { getApproximateTransactionFees } from 'auctions-core/src/fees';
+import { ETH_NUMBER_OF_DIGITS, GWEI_NUMBER_OF_DIGITS } from 'auctions-core/src/constants/UNITS';
+
+interface State {
+    baseFeePerGas: BigNumber | undefined;
+    gasParameters: GasParameters;
+    transactionFees: TransactionFees | undefined;
+}
+
+export const state = (): State => ({
+    baseFeePerGas: undefined,
+    gasParameters: {
+        maxFeePerGas: undefined,
+        maxPriorityFeePerGas: undefined,
+        gasPrice: undefined,
+    },
+    transactionFees: undefined,
+});
+
+export const getters = {
+    getBaseFeePerGas(state: State) {
+        return state.baseFeePerGas;
+    },
+    getGasParameters(state: State) {
+        return state.gasParameters;
+    },
+    getTransactionFees(state: State) {
+        return state.transactionFees;
+    },
+};
+
+export const mutations = {
+    setBaseFeePerGas(state: State, baseFeePerGas: BigNumber) {
+        state.baseFeePerGas = baseFeePerGas;
+    },
+    setGasParameters(state: State, gasParameters: GasParameters) {
+        state.gasParameters = {
+            ...state.gasParameters,
+            ...gasParameters,
+        };
+    },
+    setTransactionFees(state: State, transactionFees: TransactionFees) {
+        state.transactionFees = transactionFees;
+    },
+};
+
+export const actions = {
+    async fetchBaseFeePerGas({ commit, rootGetters }: ActionContext<State, State>) {
+        const network = rootGetters['network/getMakerNetwork'];
+        if (!network) {
+            return;
+        }
+        const baseFeePerGas = await getBaseFeePerGas(network);
+        commit('setBaseFeePerGas', baseFeePerGas.shiftedBy(GWEI_NUMBER_OF_DIGITS));
+    },
+    async fetchGasParameters({ commit, rootGetters }: ActionContext<State, State>) {
+        const network = rootGetters['network/getMakerNetwork'];
+        if (!network) {
+            return;
+        }
+        const initialGasParameters = await getGasParametersForTransaction(network);
+
+        const gasParameters: GasParameters = {
+            maxFeePerGas:
+                initialGasParameters.maxFeePerGas &&
+                new BigNumber(initialGasParameters.maxFeePerGas)
+                    .shiftedBy(-ETH_NUMBER_OF_DIGITS)
+                    .shiftedBy(GWEI_NUMBER_OF_DIGITS)
+                    .toString(),
```

Comment:
As discussed, please never use `.toString()` on big numbers. Also, my suggestion would be of course to format numbers in the view, not in the store

Reply:
I moved the formatting to the component, bypassing all this `toString` back and forth.